pandas/io/s3.py

```diff
@@ -14,17 +14,15 @@ def _strip_schema(url):
     return result.netloc + result.path
 
 
-def get_filepath_or_buffer(
-    filepath_or_buffer, encoding=None, compression=None, mode=None
-):
+def get_file_and_filesystem(filepath_or_buffer, encoding=None, mode=None):
```

Comment:
Can be done as a follow up but would be nice to annotate these while touching func

Reply:
Cool, I didn't realize pandas was using type annotations.